### PR TITLE
Small refactorings

### PR DESCRIPTION
**Pull request by mattn**
* simplify code with getenv/setenv
* use _ for avoiding modify blackhole register
* use stridx instead of =~

### CODE REVIEW

File: autoload/tinygo.vim [hunk 1]
```diff
@@ -16,80 +16,49 @@
...
-        if has('nvim')
-            call execute("LspStop")
-        else
-            call execute("LspStopServer")
+    let oldenv = {}
+    for key in ['GOROOT', 'GOOS', 'GOARCH', 'GOFLAGS']
+        let value = getenv(key)
+        if value != v:null
+            let oldenv[key] = value
+            unlet $GOROOT
```
Comment (sago35) [inline autoload/tinygo.vim]: Probably forgot to delete it.

```suggestion
```

**Review by sago35 — CHANGES_REQUESTED** [review]
Thank you very much.
I learned a great lesson from you.

I was able to understand the entire code.
It became very easy to understand.

Please correct only the part I commented on.

**Comment by mattn** [comment]
Thank your review.

**Review by sago35 — APPROVED** [review]
LGTM
Thank you.

**Pull request merged**